src/main/java/org/amahi/anywhere/activity/TestEditText.java

```diff
@@ -0,0 +1,47 @@
+package org.amahi.anywhere.activity;
+
+import android.content.Context;
+import android.support.design.widget.TextInputEditText;
+import android.util.AttributeSet;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+import android.widget.ImageView;
+
+public class TestEditText extends TextInputEditText {
+
+    private static ImageView imageView;
+    private static EditText user, password;
```

Comment:
You should avoid making objects static like this. This will create memory leak as these objects would not be destroyed even when the AuthenticationActivity and its layout are destroyed (ie after login is successful). These will retain the space in memory and their reference to the specified objects. This must avoided at all costs! 

<img width="593" alt="warning" src="https://cloud.githubusercontent.com/assets/15021142/24368203/30d38c18-133d-11e7-948c-fdf4f22a717d.png">

Read [here](http://blog.nimbledroid.com/2016/05/23/memory-leaks.html) for more details on memory leak.

A better way to do this would be by using interfaces ie by making a custom listener in the TestEditText class and implementing it in the Activity. I hope you get this!